################################################################################
SHORT DESCRIPTION: 
################################################################################
Set up VGA hooks for passed through PCI devices as well.

################################################################################
LONG DESCRIPTION: 
################################################################################
ROMs are deployed in the guest memory by hvmloader, but some hooks can help to
manage specific behaviour for PCI devices passed through to the guest.

################################################################################
CHANGELOG 
################################################################################
Documented: Eric Chanudet, chanudete@ainfosec.com, 17/03/2015
Ported to SeaBIOS 1.7.5: Eric Chanudet, chanudete@ainfosec.com, 01/03/2015

################################################################################
REMOVAL 
################################################################################
This patch is required for amd-gpu-support.patch.

################################################################################
UPSTREAM PLAN 
################################################################################
This is an OpenXT work-around. There is no plan to upstream this patch.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
Depended on by SeaBIOS: amd-gpu-support.patch

################################################################################
PATCHES
################################################################################
Index: seabios-1.9.1/src/optionroms.c
===================================================================
--- seabios-1.9.1.orig/src/optionroms.c
+++ seabios-1.9.1/src/optionroms.c
@@ -432,6 +432,14 @@ vgarom_setup(void)
     ScreenAndDebug = romfile_loadint("etc/screen-and-debug", 1);
 
     if (CONFIG_OPTIONROMS_DEPLOYED) {
+        // setup hooks for vga pci devices
+        struct pci_device *pci;
+        foreachpci(pci) {
+            if (!is_pci_vga(pci))
+                continue;
+            vgahook_setup(pci);
+            break;
+        }
         // Option roms are already deployed on the system.
         init_optionrom((void*)BUILD_ROM_START, 0, 1);
     } else {
